### PR TITLE
Fix the parser issue

### PR DESCRIPTION
**Pull request by Habush**
This fixes the parser issue #162 without depending the merging of https://github.com/aconchillo/guile-json/pull/54 unlike PR #172 . I agree with @linas & @rekado in  that what I did in #172 was dangerous and I will close the PR to the `guile-json` upstream repo.

### CODE REVIEW

File: annotation/parser.scm [hunk 1]
```diff
@@ -153,15 +153,15 @@ graph by mutating global variables."
...
+  (expr->graph expr))
 
 (define* (atomese-parser port #:optional mode)
   (set! *nodes* '())
   (set! *edges* '())
   (set! *atoms* '())
   (set! *annotation* "")
   (set! *prev-annotation* "")
-  (atomese->graph
+  (for-each atomese->graph
```
Comment (rekado) [inline annotation/parser.scm]: Shouldn't the caller of `atomese-parser` take care of mapping it over a list of atomese expressions?
It seems wrong to me to return a single graph when a list of expressions is passed as an argument.

Comment (Habush) [reply annotation/parser.scm]: The graph is used for visualization and there can be only one graph being visualized at a time for a request. Hence, we need to get a single graph from the parser.

**Comment by mjsduncan** [comment]
i tried to use this commit as the scheme module for the local/development branch of the annotation service and the docker build still halts with 4 failed unit tests.

**Comment by Habush** [comment]
@mjsduncan The tests pass for me locally and also in the CI. So can you please share the logs for the failing tests? You can find the logs under the `tests` directory.

File: annotation/parser.scm (hunk 1 above)
Comment (Habush) [reply annotation/parser.scm]: @rekado maybe you can create an issue for this so it can be tracked there?

Comment (Habush) [reply annotation/parser.scm]: @rekado The latest commit resolves this. Please check it.

**Comment by Habush** [comment]
@rekado @mjsduncan Merging this as it is needed for the production version of the annotation-service.

**Pull request merged**